src/Servers/HttpSys/test/FunctionalTests/AuthenticationTests.cs

```diff
@@ -400,6 +400,60 @@ public async Task AuthTypes_DisableAutomaticAuthentication(AuthenticationSchemes
         }
     }
 
+    [ConditionalTheory]
+    [InlineData(AuthenticationSchemes.Negotiate)]
+    [InlineData(AuthenticationSchemes.NTLM)]
+    [InlineData(AuthenticationSchemes.Negotiate | AuthenticationSchemes.NTLM)]
+    public async Task AuthTypes_EnableKerberosCredentialCaching(AuthenticationSchemes authType)
+    {
+        using (var server = Utilities.CreateDynamicHost(out var address, options =>
+        {
+            options.Authentication.Schemes = authType;
+            options.Authentication.AllowAnonymous = DenyAnoymous;
+            options.Authentication.EnableKerberosCredentialCaching = true;
+        },
+        httpContext =>
+        {
+            // There doesn't seem to be a simple way of testing the `EnableKerberosCredentialCaching`
```

Comment:
I would have guessed there was a feature exposing the properties.  @Tratcher?

Reply:
There wouldn't be any meaningful properties. You'd have to check for things like the lack of an auth header on subsequent requests, while still having an authenticated user.

Reply:
> There wouldn't be any meaningful properties. You'd have to check for things like the lack of an auth header on subsequent requests, while still having an authenticated user.

The problem with meaningfully testing this option is that it needs a multi-machine setup in the AD, so that Kerberos would kick in. Because if the client is just looping back to itself, auth will happen through NTLM.

This might [change in the future](https://techcommunity.microsoft.com/t5/windows-it-pro-blog/the-evolution-of-windows-authentication/ba-p/3926848) with the introduction of the local KDC. But for now I don't think there is a reliable way testing the full scenario, so I decided to go with a simple smoke test.

Reply:
Oh, I thought getting to the property was the hard part.  If it requires a bunch of environment setup, then fair enough.